src/features/home/pages/ThematicHome.tsx

```diff
@@ -107,6 +118,65 @@ export const ThematicHome: FunctionComponent = () => {
   const { modules, id, thematicHeader } = useHomepageData(params.homeId) || {}
   const { userLocation } = useLocation()
   const isLocated = !!userLocation
+  const { user, isLoggedIn } = useAuthContext()
+  const thematic = useMapSubscriptionHomeIdsToThematic(params.homeId)
+  const { showSuccessSnackBar } = useSnackBarContext()
+  const {
+    visible: isNotificationsModalVisible,
+    showModal: showNotificationsModal,
+    hideModal: hideNotificationsModal,
+  } = useModal(false)
+  const {
+    visible: isUnsubscribingModalVisible,
+    showModal: showUnsubscribingModal,
+    hideModal: hideUnsubscribingModal,
+  } = useModal(false)
+  const {
+    visible: isSubscriptionSuccessModalVisible,
+    showModal: showSubscriptionSuccessModal,
+    hideModal: hideSubscriptionSuccessModal,
+  } = useModal(false)
+
+  const onUpdateSubscriptionSuccess = async () => {
+    if (!thematic) return
+    const hasSubscribedTimes =
+      (await storage.readObject<number>('has_subscribed_to_thematic_times')) ?? 0
+    if (hasSubscribedTimes < 3) {
+      showSubscriptionSuccessModal()
+      await storage.saveObject('has_subscribed_to_thematic_times', hasSubscribedTimes + 1)
+    } else {
+      showSuccessSnackBar({
+        message: `Tu suis le thème “${mapSubscriptionThemeToName[thematic]}”\u00a0! Tu peux gérer tes alertes depuis ton profil.`,
+        timeout: SNACK_BAR_TIME_OUT,
+      })
+    }
+  }
+
+  const {
+    isSubscribeButtonActive,
+    isAtLeastOneNotificationTypeActivated,
+    updateSubscription,
+    updateSettings,
+  } = useThematicSubscription({
+    user,
+    thematic,
+    onUpdateSubscriptionSuccess,
+  })
+
+  const onSubscribeButtonPress = () => {
+    if (!isAtLeastOneNotificationTypeActivated) {
+      showNotificationsModal()
+    } else if (!isSubscribeButtonActive) {
+      updateSubscription()
+    } else {
+      showUnsubscribingModal()
+    }
```

Comment:
On peut regrouper les cas où on affiche les modales en premier, et le cas nominal en dernier
```suggestion
    if (!isAtLeastOneNotificationTypeActivated) {
      showNotificationsModal()
    } else if (isSubscribeButtonActive) {
      showUnsubscribingModal()
    } else {
      updateSubscription()
    }
```